cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/MetricSample.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling;
+
+import com.linkedin.cruisecontrol.metricdef.MetricDef;
+import com.linkedin.cruisecontrol.metricdef.MetricInfo;
+import com.linkedin.cruisecontrol.model.Entity;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MetricSample<G, E extends Entity<G>> {
```

Comment:
Since G is never referenced by itself it is never needed.

Reply:
This is mainly for type safety. We will pass the entity to some other method which requires the G